code/obj/critter/misc.dm

```diff
@@ -1037,30 +1037,34 @@
 		qdel(src)
 
 	CritterAttack(mob/M)
-		src.attacking = 1
-		src.visible_message("<span class='combat'><B>The [src.name]</B> starts to envelop [M]!</span>")
+		if(GET_COOLDOWN(src, "envelope_attack"))
```

Comment:
yeah, this pattern is because I was trying out some stuff with src.attacking to stop the science ghost repeatedly booping you while you were being absorbed - it didn't work out and didn't really matter anyway.